ProgrammaticUI/Controllers/HomeViewController.swift

```diff
@@ -29,49 +30,36 @@ class HomeViewController: UIViewController {
         setStackViewConstraints()
     }
 
-    func addSummaryStackToStackView() {
+    private func addSummaryStackToStackView() {
         let summaryStack = UIStackView()
         summaryStack.axis = .vertical
         summaryStack.spacing = 15
         summaryStack.distribution = .fillProportionally
 
-        let summaryLabel = UILabel()
-        summaryLabel.text = "Summary"
-        summaryLabel.font = .boldSystemFont(ofSize: 40)
-        //summaryLabel.font = UIFont(name: "AvenirNext-DemiBoldItalic", size: 48)
         summaryStack.addArrangedSubview(summaryLabel)
-
-        kmsLabel = UILabel()
-        caloriesLabel = UILabel()
-        durationLabel = UILabel()
         summaryStack.addArrangedSubview(kmsLabel)
         summaryStack.addArrangedSubview(caloriesLabel)
         summaryStack.addArrangedSubview(durationLabel)
-        //summaryStack.addArrangedSubview(imageFitness)
-        stackView.addArrangedSubview(summaryStack)
 
-        // Set initial label values (you can set your own initial values if needed)
-        kmsLabel.text = "Kms: 0.0"
-        caloriesLabel.text = "Calories: 0"
-        durationLabel.text = "Duration: 00:00:00"
+        stackView.addArrangedSubview(summaryStack)
     }
 
-    func setStackViewConstraints() {
+    private func setStackViewConstraints() {
```

Comment:
1. Please try to avoid negative constraints.

Use this case:
```swift
bar.leadingAnchor.constraint(equalTo: foo.leadingAnchor, constant: -50)
```
In favor of:
```swift
foo.leadingAnchor.constraint(equalTo: bar.leadingAnchor, constant: -50)
```

2. I don't see a `stackView.bottomAnchor` or any other vertical position anchor for the stack. This might raise a constraint conflict that the autolayout algorithm will try to solve at runtime. This might cause lag and unexpected behavior.